File: backend/src/main/java/org/example/backend/domain/tradechat/dto/TradeChatMessageDto.java
package org.example.backend.domain.tradechat.dto;

import org.example.backend.domain.tradechat.entity.TradeChatMessage;

import java.time.LocalDateTime;

public record TradeChatMessageDto(
        Long messageId,
        Long senderId,
        String senderNickname,
        String content,
        LocalDateTime sendDate
) {
    public static TradeChatMessageDto from(TradeChatMessage message) {
        return new TradeChatMessageDto(
                message.getId(),
                message.getSender().getMemberId(),
                message.getSender().getNickname(),
                message.getContent(),
                message.getSendDate()
        );
    }
}

File: backend/src/main/java/org/example/backend/domain/tradechat/dto/TradeChatMessageDto.kt
package org.example.backend.domain.tradechat.dto

import org.example.backend.domain.tradechat.entity.TradeChatMessage
import java.time.LocalDateTime

@JvmRecord
data class TradeChatMessageDto(
    val messageId: Long,
    val senderId: Long?,
    val senderNickname: String?,
    val content: String,
    val sendDate: LocalDateTime?
) {
    companion object {
        @JvmStatic
        fun from(message: TradeChatMessage): TradeChatMessageDto {
            return TradeChatMessageDto(
                message.id,
                message.sender.memberId,
                message.sender.nickname,
                message.content,
                message.sendDate
            )
        }
    }
}
